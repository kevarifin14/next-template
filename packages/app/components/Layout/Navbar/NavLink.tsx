import Link from 'next/link';

export default function NavLink({ href, children }) {
  return (
    <Link href={href}>
      <a className="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium">
        {children}
      </a>
    </Link>
  );
}
