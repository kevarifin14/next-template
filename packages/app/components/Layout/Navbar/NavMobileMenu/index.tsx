import { Transition } from '@headlessui/react';
import { MenuIcon, XIcon } from '@heroicons/react/outline';
import { useRouter } from 'next/router';
import { useState } from 'react';

import Button from 'components/Button';
import LogoImage from 'components/LogoImage';

import NavButton from './NavButton';
import NavIconLink from './NavIconLink';

export default function NavMobileMenu({ links }) {
  const [isOpen, setIsOpen] = useState(false);
  const router = useRouter();

  return (
    <>

      <div className="md:hidden">
        <NavButton onClick={() => setIsOpen(true)}>
          <MenuIcon className="h-6 w-6" />
        </NavButton>
      </div>

      <Transition
        show={isOpen}
        className="absolute z-30 top-0 inset-x-0 transition transform origin-top-right md:hidden"
        enter="duration-200 ease-out"
        enterFrom="opacity-0 scale-95"
        enterTo="opacity-100 scale-100"
        leave="duration-100 ease-in"
        leaveFrom="opacity-100 scale-100"
        leaveTo="opacity-0 scale-95"
      >
        <div className="rounded-lg shadow-lg bg-light divide-y-2 divide-gray-50">
          <div className="pt-5 pb-6 px-5 sm:pb-8">
            <div className="flex items-center justify-between">

              <div className="h-8 w-8 relative">
                <LogoImage />
              </div>

              <div className="-mr-2">
                <NavButton onClick={() => setIsOpen(false)}>
                  <XIcon className="h-6 w-6" />
                </NavButton>
              </div>

            </div>

            <div className="mt-6 sm:mt-8">
              <nav className="space-y-2">

                {links.map((props) => <NavIconLink key={props.title} {...props} />)}

                <Button type="primary" block onClick={() => router.push('/signin')}>
                  Sign Up
                </Button>

              </nav>
            </div>

          </div>
        </div>
      </Transition>
    </>
  );
}
