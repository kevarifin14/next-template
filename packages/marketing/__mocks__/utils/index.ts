import { HiOutlineDesktopComputer, HiOutlineGlobe, HiOutlineLightningBolt } from 'react-icons/hi';

export const mockEmail = 'mock@gmail.com';

export const mockTitle = 'mockTitle';
export const mockDescription = 'mockDescription';
export const mockSrc = 'mockSrc';

export const mockFeatures = [
  {
    name: 'Marketing',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineGlobe,
  },
  {
    name: 'Application',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineDesktopComputer,
  },
  {
    name: 'Design System',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineLightningBolt,
  },
];
