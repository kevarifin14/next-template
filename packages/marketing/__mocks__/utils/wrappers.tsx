import { NotificationsProvider, AppearanceProvider } from 'shared';

export const notificationsProviderWrapper = ({ children }) => (
  <NotificationsProvider>{children}</NotificationsProvider>
);

export const appearanceProviderWrapper = ({ children }) => (
  <AppearanceProvider>{children}</AppearanceProvider>
);
