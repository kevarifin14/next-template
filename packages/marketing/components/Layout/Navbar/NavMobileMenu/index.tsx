import { Transition } from '@headlessui/react';
import { useRouter } from 'next/router';
import { useState } from 'react';
import { HiMenu, HiX } from 'react-icons/hi';
import { Button } from 'shared';

import LogoImage from 'components/LogoImage';

import NavButton from './NavButton';
import NavIconLink from './NavIconLink';

export default function NavMobileMenu({ links }) {
  const [isOpen, setIsOpen] = useState(false);
  const router = useRouter();

  return (
    <>

      <div className="md:hidden">
        <NavButton onClick={() => setIsOpen(true)}>
          <HiMenu className="h-6 w-6" />
        </NavButton>
      </div>

      <Transition
        show={isOpen}
        className="absolute z-30 top-0 inset-x-0 transition transform origin-top-right md:hidden"
        enter="duration-200 ease-out"
        enterFrom="opacity-0 scale-95"
        enterTo="opacity-100 scale-100"
        leave="duration-100 ease-in"
        leaveFrom="opacity-100 scale-100"
        leaveTo="opacity-0 scale-95"
      >
        <div className="rounded-lg shadow-lg bg-light dark:bg-dark-light divide-y-2 divide-gray-50">
          <div className="p-6 space-y-6">
            <div className="flex items-center justify-between">

              <div className="h-8 w-8 relative">
                <LogoImage />
              </div>

              <div className="-mr-2">
                <NavButton onClick={() => setIsOpen(false)}>
                  <HiX className="h-6 w-6" />
                </NavButton>
              </div>

            </div>

            <nav className="space-y-2">

              {links.map((props) => <NavIconLink key={props.title} {...props} />)}

            </nav>

            <Button type="primary" block onClick={() => router.push('/signin')}>
              Sign Up
            </Button>

          </div>
        </div>
      </Transition>
    </>
  );
}
