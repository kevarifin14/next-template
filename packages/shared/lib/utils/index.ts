export function splitName(name: string) {
  return {
    firstName: name.split(" ").slice(0, -1).join(" "),
    lastName: name.split(" ").slice(-1).join(" "),
  };
}
