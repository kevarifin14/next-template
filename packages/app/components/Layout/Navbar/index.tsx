import { BookOpenIcon, UserIcon } from '@heroicons/react/outline';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { Button } from 'shared';

import LogoImage from 'components/LogoImage';

import NavLink from './NavLink';
import NavMobileMenu from './NavMobileMenu';

export default function Navbar() {
  const router = useRouter();

  const links = [
    {
      href: '/docs',
      icon: BookOpenIcon,
      title: 'Documentation',
    },
    {
      href: '/signin',
      icon: UserIcon,
      title: 'Sign In',
    },
  ];

  return (
    <div className="px-4 sm:px-6 lg:px-8 py-2 w-full bg-light z-50">
      <div className="mx-auto max-w-7xl relative flex items-center justify-between h-16">

        <Link href="/">
          <a>
            <LogoImage className="h-8 sm:h-10" />
          </a>
        </Link>

        <div className="hidden md:block space-x-8">

          {links.map(({ title, href }) => (
            <NavLink href={href}>
              {title}
            </NavLink>
          ))}

          <Button type="primary" onClick={() => router.push('/signin')}>
            Sign Up
          </Button>

        </div>

        <NavMobileMenu links={links} />

      </div>
    </div>
  );
}
