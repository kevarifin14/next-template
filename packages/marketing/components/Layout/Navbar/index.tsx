import Link from 'next/link';
import { HiInbox, HiMoon, HiSun } from 'react-icons/hi';
import { useAppearanceContext, LogoImage } from 'shared';

import NavLink from './NavLink';
import NavMobileMenu from './NavMobileMenu';

export default function Navbar() {
  const { dark, setDark } = useAppearanceContext();
  const links = [
    {
      title: 'Packages',
      href: '/packages',
      icon: HiInbox,
    },
    {
      title: 'Docs',
      href: '/docs',
      icon: HiInbox,
    },
    {
      title: 'Guides',
      href: '/guides',
      icon: HiInbox,
    },
  ];

  return (
    <div className="px-4 sm:px-6 lg:px-8 py-2">
      <div className="mx-auto relative flex items-center justify-between h-16 max-w-7xl w-full">

        <Link href="/">
          <a>
            <LogoImage className="h-8 sm:h-10" />
          </a>
        </Link>

        <div className="hidden md:flex justify-center space-x-8 flex-1">
          {links.map(({ title, href }) => (
            <NavLink key={title} href={href}>
              {title}
            </NavLink>
          ))}
        </div>

        <div className="hidden md:flex items-center justify-end space-x-8">

          <button
            onClick={() => setDark(!dark)}
            className="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
          >
            {dark ? <HiSun className="h-6 w-6" /> : <HiMoon className="h-6 w-6" />}
          </button>

        </div>

        <NavMobileMenu links={links} />

      </div>
    </div>
  );
}
