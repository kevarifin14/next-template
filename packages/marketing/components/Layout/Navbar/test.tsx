import { act, render, fireEvent } from '@testing-library/react';

import { appearanceProviderWrapper } from '__mocks__/utils/wrappers';

import Navbar from '.';

describe('Navbar', () => {
  const renderNavbar = (props = {}) => (
    render(
      <Navbar {...props} />,
      { wrapper: appearanceProviderWrapper },
    )
  );

  it('matches snapshot', () => {
    const { baseElement } = renderNavbar();
    expect(baseElement).toMatchSnapshot();
  });

  it('toggles dark mode', () => {
    const { getAllByRole, container } = renderNavbar();
    const [darkButton] = getAllByRole('button');

    act(() => {
      fireEvent.click(darkButton);
    });

    expect(container.firstChild).toHaveClass('dark');
  });
});
