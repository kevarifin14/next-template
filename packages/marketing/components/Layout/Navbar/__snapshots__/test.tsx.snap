// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<body>
  <div>
    <div
      class=""
    >
      <div
        class="transform transition-colors h-full bg-light dark:bg-dark"
      >
        <div
          class="px-4 sm:px-6 lg:px-8 py-2"
        >
          <div
            class="mx-auto relative flex items-center justify-between h-16 max-w-7xl w-full"
          >
            <a
              href="/"
            >
              <img
                alt="logo"
                class="mx-auto w-auto h-8 sm:h-10"
                src="/logo.png"
              />
            </a>
            <div
              class="hidden md:flex justify-center space-x-8 flex-1"
            >
              <a
                class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                href="/packages"
              >
                Packages
              </a>
              <a
                class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                href="/docs"
              >
                Docs
              </a>
              <a
                class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                href="/guides"
              >
                Guides
              </a>
            </div>
            <div
              class="hidden md:flex items-center justify-end space-x-8"
            >
              <button
                class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
              >
                <svg
                  class="h-6 w-6"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
                  />
                </svg>
              </button>
            </div>
            <div>
              <div
                class="md:hidden"
              >
                <button
                  class="prose dark:prose-light dark:hover:bg-dark-light hover:bg-light-dark rounded-md p-2 inline-flex items-center justify-center focus:outline-none"
                  type="button"
                >
                  <svg
                    class="h-6 w-6"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
